src/data/loadDataFrames.ts

```diff
@@ -161,6 +154,62 @@ export function loadSingleTimeseries(fields: Array<Field<string, number[]>>, ref
     table: {},
     timeseries,
   };
+  return newFeature;
+}
+
+export function loadTimeseriesWithCustomData(
+  tsField: Array<Field<string, number[]>>,
+  refId: string,
+  tableField: Array<Field<string, any>>
+): Feature | undefined {
+  const timeVector = tsField?.[0];
+  if (timeVector == null || timeVector.type !== FieldType.time) {
+    console.warn('alert-danger', [`Timeseries data - missing Time vector in ${refId}.`]);
+    return;
+  }
+  if (tableField.length == null) {
+    console.warn('alert-danger', [`No data or wrong query for custom constants table.`]);
+    return;
+  }
+
+  const valueVector = firstValueField(tsField);
+  if (valueVector == null) {
+    console.warn('alert-danger', [`Timeseries data - missing Value vector in ${refId}.`]);
+    return;
+  }
+
+  const newFeature = defaultFeature('value', '', refId);
+
+  const { t, v } = noNulls(timeVector.values as number[], valueVector.values as number[]);
+  const timeseries = {
+    time: { ...timeVector, values: t },
+    values: { ...valueVector, values: v },
+  };
+
+  // RESERVED VALUES!
+  // nominal, lsl, usl, min, max, mean, range, lcl_Rbar, ucl_Rbar, lcl_Sbar, ucl_Sbar, lcl, ucl,
+  // This values in table query are reserved for frontend calculations.
+  // If you want to use your custom database constant values use a different name in SQL query.
+  //TODO: add warning if reserved values are used
```

Comment:
I think we should check whether reserved values are used, if so - display or log some warning.

Reply:
Done in [bffe94e](https://github.com/KensoBI/spc-panel/pull/30/commits/bffe94e92119138b0e1945c1706b05257eb3efe9)